dice-game-ui-angular/src/app/game/game.component.ts

```diff
@@ -1,13 +1,334 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, HostListener } from '@angular/core';
 import * as d3 from 'd3';
 
+import { SimpleGameEngine, SimpleDice, Rect, Size } from '../../../../engine';
+
 @Component({
   selector: 'app-game',
   templateUrl: './game.component.html',
   styleUrls: ['./game.component.scss'],
 })
 export class GameComponent implements OnInit {
-  constructor() {}
+  private readonly setColor = '#D28EFF';
+  private readonly selectedColor = '#3f51b5';
+  private readonly emptyColor = 'lightblue';
+  private canvasMaxHeight = 600;
+
+  private gameEngine;
+  private simpleDice;
+
+  private points: FieldPoint[] = [];
+
+  private selecting = false;
+  private svg = null;
+  private startPoint: FieldPoint = null;
+
+  public dice1 = 0;
+  public dice2 = 0;
+  public gameFinished = false;
+  public score = 0;
+
+  constructor() {
+    let fieldSize = 6;
+    const storageFieldSize = localStorage.getItem('field-size');
```

Comment:
Make a separate service to get some settings from storage.

Reply:
Created